### PR TITLE
homework16

### PR DESCRIPTION
**Pull request by Tim-D53**
Script to create and fill database
Task:
1. Write SQL queries for table creation for a data model that you created for prev homework (Airbnb model)
2. Write 3 rows (using INSERT queries) for each table in the data model
3. Create the next analytic queries:
      1. Find a user who had the biggest amount of reservations. Return user name and user_id

### CODE REVIEW

**Comment by dvn2k** [comment]
Please add the task description to the comment on the pull request. And split the script on two: data definition and data manipulation.

**Comment by Tim-D53** [comment]
> Please add the task description to the comment on the pull request. And split the script on two: data definition and data manipulation.

Fixed

File: homwork-16_DataDefinition.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
+DROP TABLE IF EXISTS "RESERVATIONS";
+DROP TABLE IF EXISTS "ROOMS";
+DROP TABLE IF EXISTS "HOSTS";
+DROP TABLE IF EXISTS "GUESTS";
+DROP TABLE IF EXISTS "USERS";
+
+CREATE TABLE IF NOT EXISTS "USERS"
```
Comment (dvn2k) [inline homwork-16_DataDefinition.sql]: Why do you have this NOT EXISTS clause? It means that the creation of the table will be skipped if the table exists. Please fix if it is not expected behavior.

File: homwork-16_DataDefinition.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    name text NOT NULL,
+    residents smallint NOT NULL,
+    price money NOT NULL,
+    aircondition boolean NOT NULL,
+    refrigerator boolean NOT NULL,
+    CONSTRAINT rooms_id_pk PRIMARY KEY (id),
+    CONSTRAINT host_id_fk FOREIGN KEY (host_id) REFERENCES "HOSTS" (id)
+);
+
+CREATE TABLE IF NOT EXISTS "RESERVATIONS"
```
Comment (dvn2k) [inline homwork-16_DataDefinition.sql]: I suggest you fix as many assumptions as possible on the database level (be as strict as possible :). Please add a comment with a description of the purpose for each added check or constraint.  
E.g. currently a user can rent his room - is it expected behavior? If yes, I think the reservation for yourself should be excluded from the [select query](https://github.com/Tim-D53/phyton/pull/2/files#diff-8f264b82c62132c0661132b00b265fdb3308dc9437823c35a884d9d242bc061cR26)

File: homework-16_DataManipulation.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
+DELETE FROM "RESERVATIONS";
+DELETE FROM "ROOMS";
+DELETE FROM "GUESTS";
+DELETE FROM "HOSTS";
+DELETE FROM "USERS";
+
+INSERT INTO "USERS" (id, name) VALUES (1, 'Host and Guest 1');
+INSERT INTO "USERS" (id, name) VALUES (2, 'Guest 2');
+INSERT INTO "USERS" (id, name) VALUES (3, 'Host and Guest 3');
+
```
Comment (dvn2k) [inline homework-16_DataManipulation.sql:10]: Please add comments on what you meant on some of the inserts below.

File: homework-16_DataManipulation.sql [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+INSERT INTO "ROOMS" (id, host_id, name, residents, price, aircondition, refrigerator) VALUES (1, 1, 'Room 1', 1, 300, true, true);
+INSERT INTO "ROOMS" (id, host_id, name, residents, price, aircondition, refrigerator) VALUES (2, 1, 'Room 2', 1, 100, false, false);
+INSERT INTO "ROOMS" (id, host_id, name, residents, price, aircondition, refrigerator) VALUES (3, 3, 'Room 3', 1, 200, false, true);
+
+INSERT INTO "RESERVATIONS" (guest_id, room_id) VALUES (1, 3);
+INSERT INTO "RESERVATIONS" (guest_id, room_id) VALUES (2, 1);
+INSERT INTO "RESERVATIONS" (guest_id, room_id) VALUES (2, 2);
+
+SELECT u.id AS user_id, u.name AS user_name
```
Comment (dvn2k) [inline homework-16_DataManipulation.sql:27]: The query returns any user with the biggest amount of reservations. Could you please add the query to return all users with the biggest amount of reservations?

File: homwork-16_DataDefinition.sql (hunk 1 above)
Comment (Tim-D53) [reply homwork-16_DataDefinition.sql]: Removed.

File: homwork-16_DataDefinition.sql (hunk 2 above)
Comment (Tim-D53) [reply homwork-16_DataDefinition.sql]: Some restrictions added and commented. I so not know how to easily reastrict "a user can rent his room" on DB level. Let's decide that "a user rent his room" means that user stop rent-out the room :). I will fix the select query to take this into account.

File: homework-16_DataManipulation.sql (hunk 3 above)
Comment (Tim-D53) [reply homework-16_DataManipulation.sql:10]: Done.

File: homework-16_DataManipulation.sql (hunk 4 above)
Comment (Tim-D53) [reply homework-16_DataManipulation.sql:27]: Sorry, but it is out of scope. The request was about one user :)

**Review by dvn2k — APPROVED** [review]
Strictly speaking, you do not add 3 rows into each table (see GUESTS and HOSTS). But I think the solution is good enough to be accepted.

**Pull request merged**